fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/group/ChunkReader.java

```diff
@@ -0,0 +1,328 @@
+import com.ibm.fhir.persistence.FHIRPersistence;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContext;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContextFactory;
+import com.ibm.fhir.persistence.helper.FHIRPersistenceHelper;
+import com.ibm.fhir.persistence.helper.FHIRTransactionHelper;
+import com.ibm.fhir.search.compartment.CompartmentUtil;
+import com.ibm.fhir.search.context.FHIRSearchContext;
+import com.ibm.fhir.search.util.SearchUtil;
+
+/**
+ * Bulk patient group export Chunk implementation - the Reader.
+ */
+public class ChunkReader extends AbstractItemReader {
+    int pageNum = 1;
+    // List for the patients
+    List<Member> patientMembers = null;
+    private final static Logger logger = Logger.getLogger(ChunkReader.class.getName());
+    int indexOfCurrentResourceType = 0;
+    // Control the number of records to read in each page.
+    int pageSize = Constants.DEFAULT_SEARCH_PAGE_SIZE;
+
+    private FHIRPersistence fhirPersistence;
+    private List<String> resourceTypes;
+
+    /**
+     * Fhir tenant id.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.tenant")
+    String fhirTenant;
+
+    /**
+     * Fhir data store id.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.datastoreid")
+    String fhirDatastoreId;
+
+    /**
+     * Fhir ResourceType.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.resourcetype")
+    String fhirResourceType;
+
+    /**
+     * Fhir Search from date.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.fromdate")
+    String fhirSearchFromDate;
+
+    /**
+     * Fhir search to date.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.todate")
+    String fhirSearchToDate;
+
+    /**
+     * Fhir search patient group id.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.patientgroupid")
+    String fhirSearchPatientGroupId;
+
+    /**
+     * Fhir search page size.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.pagesize")
+    String fhirSearchPageSize;
+
+    @Inject
+    JobContext jobContext;
+
+    public ChunkReader() {
+        super();
+    }
+
+    private void fillChunkDataBuffer(List<Member> patientRefs) throws Exception {
+        TransientUserData chunkData = (TransientUserData) jobContext.getTransientUserData();
+        int compartmentPageNum = 1;
+        int resSubTotal = 0;
+        FHIRSearchContext searchContext;
+        Class<? extends Resource> resourceType = ModelSupport
+                .getResourceType(resourceTypes.get(indexOfCurrentResourceType));
+        if (chunkData != null) {
+            for (Member patientRef : patientRefs) {
+                if (patientRef == null) {
+                    continue;
+                }
+
+                String patientId =  patientRef.getEntity().getReference().getValue().substring(8);
+                Map<String, List<String>> queryParameters = new HashMap<>();
+
+                List<String> searchCreterial = new ArrayList<String>();
```

Comment:
```suggestion
                List<String> searchCriteria = new ArrayList<String>();
```

Reply:
this is a refactor, will change in eclipse and commit.